test/unit_tests/tracking/EventTrackingRepositorySpec.js

```diff
@@ -24,18 +24,14 @@
 describe('z.tracking.EventTrackingRepository', () => {
   const test_factory = new TestFactory();
 
-  beforeEach(done => {
-    test_factory
-      .exposeTrackingActors()
-      .then(() => {
-        TestFactory.tracking_repository._is_domain_allowed_for_tracking = () => true;
-        done();
-      })
-      .catch(done.fail);
+  beforeEach(() => {
+    return test_factory.exposeTrackingActors().then(() => {
+      TestFactory.tracking_repository._is_domain_allowed_for_tracking = () => true;
+    });
```

Comment:
Learned something new here. Didn't know that jasmine gained proper Promise in version 2.7 support making the `done` schema obsolete.

https://jasmine.github.io/tutorials/async

Reply:
I already removed a few in some of our tests.
The cool thing is, you don't need to do `.catch(done.fail)` if you return the promise to `jasmine`.
So no more silent error when you forget that line :)